python-sdk/nuscenes/map_expansion/arcline_path_utils.py

```diff
@@ -186,19 +190,97 @@ def discretize(arcline_path_3: Dict[str, Any],
     return discretization
 
 
-def discretize_lane(arcline_list: List[Dict[str, Any]],
+def discretize_lane(lane: List[ArcLinePath],
                     resolution_meters: float) -> List[Tuple[float, float, float]]:
     """
     Discretizes a lane and returns list of all the poses alone the lane.
-    :param arcline_list: Lanes are represented as a list of arcline_3_paths.
+    :param lane: Lanes are represented as a list of arcline_3_paths.
     :param resolution_meters: How finely to discretize the lane. Smaller values ensure turning
         lanes are properly represented.
     :return: List of pose tuples along the lane.
     """
 
     pose_list = []
-    for arcline_path in arcline_list:
-        poses = discretize(arcline_path, resolution_meters)
+    for path in lane:
+        poses = discretize(path, resolution_meters)
         for pose in poses:
             pose_list.append(pose)
     return pose_list
+
+
+def length_of_lane(lane: List[ArcLinePath]) -> float:
+    """
+    Calculates the length of a lane
+    :param lane: Lane
+    :return: Length of lane
+    """
+
+    return sum(sum(path['segment_length']) for path in lane)
+
+
+def project_pose_to_lane(pose: Pose, lane: List[ArcLinePath]) -> Tuple[Pose, float]:
+    """
+    Find the closest pose on a lane to a query pose and additionally return the
+    distance along the lane for this pose. Note that this function does
+    not take the heading of the query pose into account.
+    :param pose: Query pose
+    :param lane: Will find the closest pose on this lane
+    :return: Tuple of the closest pose and the distance along the lane
+    """
+
+    discretized_lane = discretize_lane(lane, resolution_meters=0.5)
```

Comment:
Let's let the user control the resolution. Say make it a default param?

Reply:
Sure thing.